kie-server-parent/kie-server-tests/kie-server-integ-tests-all/src/test/java/org/kie/server/integrationtests/jbpm/cases/CaseRuntimeDataServiceIntegrationTest.java

```diff
@@ -2381,6 +2455,114 @@ public void testFindCaseTasksAssignedAsStakeholderPagination() {
+        assertNotNull(tasks);
+        assertEquals(1, tasks.size());
+        assertEquals("Hello2", tasks.get(0).getName());
+        assertEquals(potOwner, tasks.get(0).getActualOwner());
+        assertEquals(Reserved, Status.valueOf(tasks.get(0).getStatus()));
+        assertEquals("User's comment", tasks.get(0).getDescription());
+        taskClient.startTask(CONTAINER_ID, tasks.get(0).getId(), potOwner);
+        taskClient.completeTask(CONTAINER_ID, tasks.get(0).getId(), potOwner, null);
+
+        taskInput = new HashMap<>();
+        taskInput.put("GroupId", CASE_HR_GROUP);
+        taskInput.put("Comment", "HR's comment");
+        caseClient.triggerAdHocFragment(CONTAINER_ID, caseId, "Hello2", taskInput);
+        tasks = caseClient.findCaseTasksAssignedAsPotentialOwner(caseId, potOwner, 0, 10);
+        assertNotNull(tasks);
+        if (isInHR) {
+            assertEquals(1, tasks.size());
+            assertEquals("Hello2", tasks.get(0).getName());
+            assertNull(tasks.get(0).getActualOwner());
+            assertEquals(Ready, Status.valueOf(tasks.get(0).getStatus()));
+            assertEquals("HR's comment", tasks.get(0).getDescription());
+            taskClient.claimTask(CONTAINER_ID, tasks.get(0).getId(), potOwner);
+            taskClient.startTask(CONTAINER_ID, tasks.get(0).getId(), potOwner);
+            taskClient.completeTask(CONTAINER_ID, tasks.get(0).getId(), potOwner, null);
+        } else {
+            assertEquals(0, tasks.size());
+        }
+    }
+
+    private void assertTaskListAsBusinessAdmin(String caseId, String caseOwner, String lookupUser) throws Exception {
+        List<TaskSummary> tasks = caseClient.findCaseTasksAssignedAsBusinessAdministrator(caseId, USER_ADMINISTRATOR, 0, 10);
+        assertNotNull(tasks);
+        assertEquals(1, tasks.size());
+
+        Map<String, Object> taskInput = new HashMap<>();
+
+        changeUser(caseOwner);
+        taskInput.put("BusinessAdministratorId", "contact");
+        taskInput.put("Comment", "User's comment");
+        caseClient.triggerAdHocFragment(CONTAINER_ID, caseId, "Hello2", taskInput);
+
+        changeUser(lookupUser);
+        tasks = caseClient.findCaseTasksAssignedAsBusinessAdministrator(caseId, USER_ADMINISTRATOR, 0, 10);
+        assertNotNull(tasks);
+        assertEquals(2, tasks.size()); // 1 for Yoda and 1 for John
+        tasks.stream()
+                .filter(taskSummary -> USER_JOHN.equals(taskSummary.getActualOwner()))
+                .forEach(task -> {
+                    assertEquals("Hello2", task.getName());
+                    assertEquals(Reserved, Status.valueOf(task.getStatus()));
+                    assertEquals("User's comment", task.getDescription());
+                    taskClient.startTask(CONTAINER_ID, task.getId(), USER_JOHN);
+                    taskClient.completeTask(CONTAINER_ID, task.getId(), USER_JOHN, null);
+                });
```

Comment:
You could also check the other task is assigned to Yoda

Reply:
Done